component/file_cache/file_cache.go

```diff
@@ -1268,11 +1325,18 @@ func (fc *FileCache) TruncateFile(options internal.TruncateFileOptions) error {
 	} else {
 		// If size is not 0 then we need to open the file and then truncate it
 		// Open will force download if file was not present in local system
+
 		h, err = fc.OpenFile(internal.OpenFileOptions{Name: options.Name, Flags: os.O_RDWR, Mode: fc.defaultPermission})
+		if err != nil {
+			log.Err("FileCache::TruncateFile : Error calling OpenFile with %s [%s]", options.Name, err.Error())
+		}
+
+		h, err = fc.DownloadFile(internal.DownloadFileOptions{Name: options.Name, Handle: h})
```

Comment:
Since the handle doesn't change anymore, `DownloadFile` doesn't need to return it.

Reply:
`DownlaodFile `will be receiving an empty handle here (that came from `OpenFile `and needs to return the handle with updated information.

Reply:
The handle is a pointer, so `h` in the current scope is already updated when `DownloadFile` writes to it.

Reply:
Thanks. Getting used to pointers